hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/ExpressionEvaluators.java

```diff
@@ -156,12 +156,20 @@ public static Evaluator fromExpression(CallExpression expr) {
   public interface Evaluator extends Serializable {
 
     /**
-     * Decides whether it's possible to match based on the column stats.
+     * Evaluates whether it's possible to match based on the column stats.
      *
      * @param columnStatsMap column statistics
-     * @return
+     * @return false if it's not possible to match, true otherwise.
      */
     boolean eval(Map<String, ColumnStats> columnStatsMap);
+
+    /**
+     * Evaluates whether it matches based on the column values.
+     *
+     * @param columnValues column values
+     * @return true if it's matches, false otherwise.
+     */
+    boolean eval(Object[] columnValues);
   }
```

Comment:
What do you indicate for these `columnValues`? Are they come from the constant literal? Can we just wrap these constant as another `ColumnStats` and reuse the existing interfaces?

Reply:
`ColumnValues` comes from literal values. 
It's not good to wrap those value as `ColumnStats ` because they are totally different thing. For `ColumnStats`, `evaluates` method returns false if there is no any possible to match, it's just a best effort estimate.
For `ColumnValues`, `evaluates` method true if it's matches, it's an exact matching rule based on exact values.
It's clear to keep two API method.

Reply:
Not sure whether I got your idea, for a expression `Evaluator`, we firstly bind the field reference to it, that means you can then fetch the field value from the reference index and the given rows, for partition pruning, things are a little different because in on partition, all the data files may share the same partition path values, but the partition values should be also included in the data row, so you can still do the similar filteiring just like we do to data skipping.

Reply:
Yes, for partition pruning, statistics of partition columns for a specified partition is an exactly value instead of a range with [min, max] and nullCnt.
Of course, we could wrap this exact value in a `columnstats` object, which min is same with max and nullCnt is 0.
But I prefer not to do so because it would cause unnecessary overhead. For example, for `Equals`, `NotEqualsTo`, `In`, there is no need to compare the literal value with minValue and maxValue twice, just compare it with exact value.

Reply:
Leave them in two different eval method is clean.

Reply:
Sorry, let's not introduce uncecessary interfaces, the performance gains should be little.